tests/unit/parsers/test.manifestjson.js

```diff
@@ -3185,4 +3185,337 @@ describe('ManifestJSONParser', () => {
+              gecko: {
+                strict_min_version: '90.0',
+              },
+            },
+            permissions: ['ALARMS'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'unrelated permission',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: ['bookmarks'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'empty permissions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: [],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'no permissions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: undefined,
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'multiple permissions but strict_min_version > min version',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '89.0',
+              },
+            },
+            permissions: ['bookmarks', 'alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'no restricted permissions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '87.0',
+              },
+            },
+            permissions: ['bookmarks', 'alarms'],
+          },
+          restrictedPermissions: new Map([]),
+        },
+      ],
+      [
+        'versions contain non-numeric chars',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '1.1pre',
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '1.1c']]),
+        },
+      ],
+    ])(
+      'does not report an error when: %s',
+      (title, { manifestProps, restrictedPermissions }) => {
+        const { linter } = validate({
+          manifestProps,
+          restrictedPermissions,
+        });
+        const { errors } = linter.collector;
+
+        // We cannot assert `parser.isValid` because some (unrelated) warnings
+        // could set this prop to `false`.
+        expect(errors.length).toEqual(0);
+      }
+    );
+
+    it.each(['1', 'a'])(
```

Comment:
as anticipated in another review comment, I think we should also add a non string value, to be sure we have at least one test that would fail if we are not stringifying the strict_min_version value got from the parsed (but not necessarily valid) manifest:

e.g.
```
 it.each(['1', 'a', { notAString: true }])(...
```